### ISSUE TITLE
add documentation of recommended project structure for multi-platform projects

### ISSUE THREAD

**Issue by twatzl** [JetBrains/compose-multiplatform#234]
Hi,

I am trying to set up a new project with compose. For now I just want to work on a desktop app, but I want to organize my project according to the recommended defaults. However I don't quite understand the defaults in the example apps.

On the root level there is always 3 modules: android, desktop and common. However android and desktop is nothing more than the main method and a manifest. This is fine, but what I don't understand is why there is androidMain and desktopMain folders in the common module. Intuitively as someone with little prior knowledge I would expect the androidMain and desktopMain stuff to be in the respective android and desktop module.

I think it would make sense to add some documentation explaining the "ideal" or "recommended" project setup and the reasoning behind it.

It's also not really covered in the tutorials as the getting started documentation only shows how to use the wizard, but the created project structure is different from what can be seen in the example apps.

I hope I did not miss some documentation somewhere.
Best Tobi

**Comment by igordmn** [comment]
The reason why we added three modules (common, android, desktop) instead of one was because in the past we used "application" plugin which isn't compatible with "com.android.application". Now maybe we can use only one module, but we have not yet check if everything works with this approach.

There are androidMain and desktopMain in the common folder because they contain platform-specific code that is used by common module (not by android/desktop modules)

See also https://github.com/JetBrains/compose-jb/issues/121. After we resolve this issue we can think about how to add information to the tutorials about recommended way to organize multiplatform project.
